sdk/java/src/main/java/com/gojek/feast/BackOffSupplier.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.gojek.feast;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/** Backoff Supplier wraps a existing {@link Supplier} with backoff retries. */
+public class BackOffSupplier<T> {
```

Comment:
I am having second thoughts about adding all this code into Feast. Can we leave this up to the user? What value does this bring to us?

Reply:
The value it gives us is that we can greatly clean up the end to end tests, which have copy pasted implementations of poll implementations. Now that Feast SDKs supports waiting internally, we can remove the unsightly polling from end to end tests.

Reply:
Removed.